.ng-dev/tsconfig.json

```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../tsconfig.json",
   "compilerOptions": {
+    "module": "Node16",
+    "moduleResolution": "Node16",
     "noEmit": true
   },
-  "include": ["**/*.ts"],
+  "include": ["**/*.mts"],
```

Comment:
Additional note: `.mts` avoids relying on the scoping `package.json` file and this seems to be more reliable for the language service and it also makes it very explicit that this is supposed to run as ESM.

(I plan on doing the same for `ts_library` in the future -- Bazel will not be happy with scoping `package.json` files)